libtextsecure/sendmessage.js

```diff
@@ -420,9 +420,17 @@ MessageSender.prototype = {
         s => s.number === number
       );
 
+      let keysFound = false;
+      // If we don't have a session but we already have prekeys to
+      // start communication then we should use them
+      if (!haveSession && !options.isPublic) {
+        keysFound = await outgoing.getKeysForNumber(number, []);
+      }
+
       if (
         number === ourNumber ||
         haveSession ||
+        keysFound ||
         options.isPublic ||
         options.messageType === 'friend-request'
```

Comment:
"||" is probably what we want here but my god that's a lot of conditions to succeed on. Not sure I have a suggestion here just marveling at it.

Reply:
Hahaha, i'm not sure what we could do to clean this up. Most of it is descriptive anyway :(